[CAS-1421] Docs for avatar customization (#2760)
GetStream/stream-chat-android: docusaurus/docs/Android/03-ui/03-chatui.mdx

}
```

### Customizing Avatar

#### Customizing Avatar Using Styles

> You can configure avatar shape, border width, online indicator and other aspects using [AvatarStyle](https://github.com/GetStream/stream-chat-android/blob/main/stream-chat-android/src/main/java/com/getstream/sdk/chat/view/messages/AvatarStyle.kt). Changing the shape and border radius, it is possible to create this kind of avatar: 

Reviewer: ```suggestion
You can configure avatar shape, border width, online indicator and other aspects using [AvatarStyle](https://github.com/GetStream/stream-chat-android/blob/main/stream-chat-android/src/main/java/com/getstream/sdk/chat/view/messages/AvatarStyle.kt). You can create this kind of avatar by changing the shape and border-radius: 
```

![Light_mode](../assets/square_avatar.png)

> It is also possible to change the background and the picture using `AvatarBitmapFactory`.

Reviewer: ```suggestion
You can also change the background and the picture using `AvatarBitmapFactory`.
```

#### Customizing Avatar AvatarBitmapFactory

Overriding the `AvatarBitmapFactory` allows you to add custom logic for the channel's and user's avatars displayed using `AvatarView`. 

The factory has a pair of methods for both user and channel avatars, such as `createUserBitmap` and `createDefaultUserBitmap`. The first should load the avatar from the network, this is attempted first when an avatar is requested. If the first method fails, the `Default` variant is called as a fallback. This should have an implementation that never fails, and always returns a valid bitmap.

:::note
The methods mentioned above are suspending functions. If you don't use coroutines, you'll find variants with `Blocking` in their name that you can use instead.
:::

> **If you don't want to use avatars from internet and only use the default avatar**, you can call `createDefaultUserBitmap` inside `createUserBitmap` in your custom implementation of `AvatarBitmapFactory`

Reviewer: ```suggestion
**If you don't want to use avatars from the internet and only use the default avatar**, you can call `createDefaultUserBitmap` inside `createUserBitmap` in your custom implementation of `AvatarBitmapFactory`
```

```
public open suspend fun createUserBitmap(
        user: User,
        style: AvatarStyle,
        @Px avatarSize: Int,
    ): Bitmap? {
        return createDefaultUserBitmap(user, style, avatarSize)
    }
```

> **If you only would like  to change the gradient colors for the avatar**, you can use `stream_ui_avatar_gradient_colors`. You can override this color in your color xml file like: 

Reviewer 2: ```suggestion
**If you only would like to change the gradient colors for the avatar**, you can use `stream_ui_avatar_gradient_colors`. You can override this color in your color xml file like: 
```

Reviewer: ```suggestion
**If you only would like to change the gradient colors for the default avatar**, you can use `stream_ui_avatar_gradient_colors`. You can override this color in your color xml file like: 
```

}
```

### Customizing Avatar

> The SDK uses `AvatarView` in the lists of channels, users and in the message lists. The avatar is a small picture that identifies an specific channel or user. 

Reviewer: ```suggestion
The `AvatarView` is used in the lists of channels, users, and messages. The avatar is a small picture that identifies a specific channel or user. 
```

> For `Channel`, it is used the property Channel.image and for User, it is used `User.image`. The image in the AvatarView, will be the image's url in those fields: 

Reviewer: ```suggestion
The image in the `AvatarView` is being displayed based on `image` property, present in both `Channel` and `User` objects:
```

![Default avatar](../assets/default_channel.png)

> If there's no url to be displayed, the SDK falls back to a gradient color with initials for the channel or user: 

Reviewer: ```suggestion
The `AvatarView` will use default gradient color with initials if the `image` property cannot be loaded:
```

![Light_mode](../assets/square_avatar.png)

You can also change the background and the picture using `AvatarBitmapFactory`.

#### Customizing Avatar Using AvatarBitmapFactory

Overriding the `AvatarBitmapFactory` allows you to add custom logic for the channel's and user's avatars displayed using `AvatarView`. 

The factory has a pair of methods for both user and channel avatars, such as `createUserBitmap` and `createDefaultUserBitmap`. The first should load the avatar from the network, this is attempted first when an avatar is requested. If the first method fails, the `Default` variant is called as a fallback. This should have an implementation that never fails, and always returns a valid bitmap.

:::note
The methods mentioned above are suspending functions. If you don't use coroutines, you'll find variants with `Blocking` in their name that you can use instead.
:::

> **If you don't want to use avatars from the internet and only use the default avatar**, you can call `createDefaultUserBitmap` inside `createUserBitmap` in your custom implementation of `AvatarBitmapFactory`

Reviewer: ```suggestion
**If you don't want to use avatars from the internet and only use the default avatar**, you can call `createDefaultUserBitmap` inside `createUserBitmap` in your custom implementation of `AvatarBitmapFactory`:
```

```
public open suspend fun createUserBitmap(
        user: User,
        style: AvatarStyle,
        @Px avatarSize: Int,
    ): Bitmap? {
        return createDefaultUserBitmap(user, style, avatarSize)
    }
```

> This way use default avatar with gradient colors and initials for the users.

Reviewer: I think we don't need it. You already described it above the snippet
```suggestion
```

**If you only would like to change the gradient colors for the default avatar**, you can use `stream_ui_avatar_gradient_colors`.

> The normal color set include a variaty of colors: 

Reviewer: ```suggestion
The default color set includes a variety of colors: 
```

![Colorful avatars](../assets/colorful_avatars.png)

> You can override this color in your color xml file. And expand of reduce the number of supported color, like the example

Reviewer: ```suggestion
The set can be overridden in the `color.xml` file - you can expand or reduce the number of supported colors, like in the example below:
```
